### ISSUE TITLE
CERRA regional reanalysis

### ISSUE THREAD

**Issue by larsbuntemeyer** [euro-cordex/joint-evaluation#19]
CERRA regional reanalysis (5.5km) is already available in the C3S CDS for the period 1 September 1984 – 30 June 2021.

Should we use this for evaluation? The only problem i see is that it's always complicated to get data out of CDS in a repproducible way, urrghh.. 😣

**Comment by larsbuntemeyer** [comment]
@JavierDiezSierra do you have some more experience with CDS? If we decice to use CERRA data, we might need a script that queries the [cdsapi](https://github.com/ecmwf/cdsapi) for the datasets we need. I would prefer this instead of downloading it through the webapi in case we want to reproduce this.

**Comment by JavierDiezSierra** [comment]
@larsbuntemeyer Yes, no problem with that. I have broad experience with the [cdsapi](https://github.com/ecmwf/cdsapi)

**Comment by larsbuntemeyer** [comment]
[weather-tools](https://github.com/google/weather-tools) might be worth a try, seems to be quite easy to use.

**Comment by JavierDiezSierra** [comment]
Thanks @larsbuntemeyer, I'll take a look. Downloading CERRA with the CDS API is simple as long as the product information required is "analysis" and not "forecasts." The "forecast" is necessary when hourly information is needed to calculate, for example, tasmax or tasmin. In any case, it's something we've already dealt with ;-)

**Comment by gnikulin** [comment]
@JavierDiezSierra as I see one can only  download 3-hr CERRA using CDS API (large files), any experience with remote calculation of daily means when retrieving CERRA?

**Comment by JavierDiezSierra** [comment]
@gnikulin The analysis product is 3-hourly. If you need hourly data, for example, to calculate tasmax, you have to use the forecast data.

For analysis variables (non-accumulated ones), remote calculation is easier since you only need to download the 3-hourly data and resample it to monthly (e.g., tas, slp). For accumulated variables, at least three downloads (forecasts) are recommended to avoid the spin-up effect, according to the CARRA documentation [1] (the same applies to CERRA).

We conducted several analyses, and in reality, the spin-up effect is not very evident, but that's another discussion.

We have already implemented the CERRA download for the ATLAS. Given some time, I could take care of this if needed. The download is quite heavy. The recommended approach would be to download the sub-daily data (180 GB for analysis variables and 424 GB for forecast) and resample it on the fly. However, we performed this process in two steps, keeping the hourly data since the download is also very slow (a bit over five days for one forecast variable) in case any issues arise. However, I understand that in this case, storing the hourly data may not be feasible.

[1] https://confluence.ecmwf.int/pages/viewpage.action?pageId=453319309

**Comment by gnikulin** [comment]
by the way there is a recommendation to use [precipitation from CERRA-Land](https://confluence.ecmwf.int/display/CKB/CERRA%3A+Strengths+and+weaknesses#CERRA:Strengthsandweaknesses-DailyprecipitationinCERRA-Land)

**Comment by larsbuntemeyer** [comment]
I tried but i don't get [CERRA from cdsapi](https://github.com/euro-cordex/joint-evaluation/blob/main/code/data-retrieve/retrieve-cerra.py) due to a `MarsNoDataError`. I opened a ticket.

**Comment by JavierDiezSierra** [comment]
@larsbuntemeyer @gnikulin In the context of the Copernicus Atlas, we calculate monthly values of temperature and precipitation on a regular grid (~6 km) using conservative interpolation. We can re-interpolate the files to the CORDEX native to the figures of the paper (spatial an regional bias). The data is not public yet, but it will be released in the coming weeks. In the meantime, we can upload the files to the JSC and include the pipeline (downloaded from the CDS) once they become public.

My idea is to compare E-OBS with CERRA to provide an indication of the uncertainty associated with the observational datasets. We will include more datasets once we decide which ones are most suitable

Do you agree?

**Comment by gnikulin** [comment]
> I tried but i don't get [CERRA from cdsapi](https://github.com/euro-cordex/joint-evaluation/blob/main/code/data-retrieve/retrieve-cerra.py) due to a `MarsNoDataError`. I opened a ticket.

I think the problem is that you are trying to retrieve a huge amount of data also mixing the analysis and forecast (many different lead times) products. As we discussed above CERRA doesn't provide monthly means. As a test, you can try to retrieve "2m_temperature" for one year or one month, that should work.

**Comment by gnikulin** [comment]
I already mentioned that CERRA-Land should be used for precipitation (available as daily data) since CERRA really overestimates precipitation.

**Comment by JavierDiezSierra** [comment]
@gnikulin @larsbuntemeyer @jesusff 

I've already downloaded temperature data for CERRA and precipitation for CERRA-Land:

```
/mnt/CORDEX_CMIP6_tmp/aux_data/cerra-land/day/pr
/mnt/CORDEX_CMIP6_tmp/aux_data/cerra/day/tas/
```
Precipitation is downloaded directly at daily resolution, while for temperature, I aggregated the data to daily and removed the 3-hourly values.

I'll commit the script once the download is finished.

**Comment by larsbuntemeyer** [comment]
Thanks! You can overwrite [retrieve script](https://github.com/euro-cordex/joint-evaluation/blob/main/code/data-retrieve/retrieve-cerra.py), that didn't work anyway!
